packages/core/directives/map.ts

```diff
@@ -257,12 +250,12 @@ export class AgmMap implements OnChanges, OnInit, OnDestroy {
   /**
    * Options for the Fullscreen control.
    */
-  @Input() fullscreenControlOptions: FullscreenControlOptions;
+  @Input() fullscreenControlOptions: google.maps.FullscreenControlOptions;
 
   /**
    * The map mapTypeId. Defaults to 'roadmap'.
    */
-  @Input() mapTypeId: 'roadmap' | 'hybrid' | 'satellite' | 'terrain' | string = 'roadmap';
+  @Input() mapTypeId: google.maps.MapTypeId | keyof typeof google.maps.MapTypeId = 'roadmap' as google.maps.MapTypeId;
```

Comment:
This is incorrect. The **key** MapTypeId is ROADMAP, the **value** is `'roadmap'`. This is the correct definition.

See: https://developers-dot-devsite-v2-prod.appspot.com/maps/documentation/javascript/reference/map#MapTypeId

Reply:
(`keyof typeof` means use the value of an enum)

Reply:
yep, that is what I said. key is ROADMAP, value is 'roadmap'. since the doc specifically mentions 'roadmap', we should support the lowercase which is *NOT* the key

Reply:
`keyof typeof` means use the key of the enum, not the value. See [example](https://www.typescriptlang.org/play/?ts=3.5.1#code/KYOwrgtgBAsghgBwCoE8HAJIBMoG8BQUUAEgJoBCAShgCJQC8UARABYoBGATgJZZMA0hKJQDyAQRowxABQbNOAezhYIiAUIDKYpAFEAMnoy65TAM5wALsAA217lfVFdlSmIwA5E1c6c43EEz4AL74+NbAFlDAAFxQANbAKAoAZlAWaMApsIio6NhyAORsXLwFANxAA). Notice the red line under `e`

Reply:
sorry, yes. I replied too quickly and was in a meeting. I updated this to be inline with the other places we use the keys or values for enums.